django/http/multipartparser.py

```diff
@@ -744,15 +743,13 @@ def parse_header(line):
 
 
 def _parse_header_params(s):
-    plist = []
     while s[:1] == b";":
         s = s[1:]
         end = s.find(b";")
-        while end > 0 and s.count(b'"', 0, end) % 2:
+        while end > 0 and (s.count(b'"', 0, end) - s.count(b'\\"', 0, end)) % 2:
```

Comment:
My [request](https://github.com/django/django/pull/15718#issuecomment-1139260076) in #15718 was to start with PR only with this single line change (https://github.com/python/cpython/commit/1ef0c0349e8fdb5415e21231cb42edbf232b742a) and a proper regression test for it. Please revert other changes and add a regression test for this addition.

Reply:
Thank you so much for the review; also, your patience 👍.

I have made the requested changes, please let me know if there is anything else that I can do.